tests/ir/lang/test_qasm_decode.py

```diff
@@ -296,6 +297,29 @@ def test_include_simple(self) -> None:
         assert circuit.get_unitary().get_distance_from(gate_unitary) < 1e-7
 
 
+class TestReset:
+    def test_reset_single_qubit(self) -> None:
+        input = """
+            OPENQASM 2.0;
+            qreg q[1];
+            reset q[0];
+        """
+        circuit = OPENQASM2Language().decode(input)
+        expected = Reset()
+        assert circuit[0, 0].gate == expected
+
+    def test_reset_register(self) -> None:
+        input = """
+            OPENQASM 2.0;
+            qreg q[2];
+            reset q;
+        """
+        circuit = OPENQASM2Language().decode(input)
+        expected = Reset()
+        assert circuit[0, 0].gate == expected
+        assert circuit[0, 1].gate == expected
+
+
```

Comment:
We need to add tests for measurement in the middle, as this was a new PR feature.